blockreassurance.php

```diff
@@ -166,15 +237,37 @@ public function getContent()
                 $reassurance->save();
 
                 if (isset($_FILES['image']) && isset($_FILES['image']['tmp_name']) && !empty($_FILES['image']['tmp_name'])) {
-                    if ($error = ImageManager::validateUpload($_FILES['image'])) {
+
+                    $error = ImageManager::validateUpload($_FILES['image']);
+
+                    if ($error) {
                         return false;
-                    } elseif (!($tmpName = tempnam(_PS_TMP_IMG_DIR_, 'PS')) || !move_uploaded_file($_FILES['image']['tmp_name'], $tmpName)) {
+                    }
+
+                    if (!($tmpName = tempnam(_PS_TMP_IMG_DIR_, 'PS'))) {
                         return false;
-                    } elseif (!ImageManager::resize($tmpName, dirname(__FILE__).'/img/reassurance-'.(int)$reassurance->id.'-'.(int)$reassurance->id_shop.'.jpg')) {
+                    }
+
+                    if (!move_uploaded_file($_FILES['image']['tmp_name'], $tmpName)) {
+                        return false;
+                    }
+
+                    $destination = sprintf(
+                        '%s/img/reassurance-%d-%d.jpg',
```

Comment:
Haven't you changes the images directory with `views/img`?

Reply:
I hesitate, because there are 2 types of images:
1) images used for the template (`img`)
2) images uploaded by the customer (now in `views/img/`)

In usecase 1) it should be committed into git and in usecase 2) it could be an empty `img` folder with its content ignored in the `.gitignore`.

Does this look relevant ?